lib/arduino-upload.coffee

```diff
@@ -139,29 +154,39 @@ module.exports = ArduinoUpload =
 		isArduino = fs.existsSync file
 		return {isArduino, workpath, file, name}
 	_build: (options, callback, onerror, port = false) ->
+		serialWasOpen = false
+		autosaveConf = atom.config.get('arduino-upload.autosave')
+		if (autosaveConf == 'save') || (autosaveConf == 'save+reopen')
+			if autosaveConf == 'save+reopen'
+				if serial != null
+					serialWasOpen = true
+				else
```

Comment:
now this else block is kinda obsolete, also allowing to merge the `serial != null` with its parent if-condition

Reply:
Right, I should have caught that. Updating now.